packages/core/typescript/TypeScriptFormattingService.ts

```diff
@@ -0,0 +1,217 @@
+      ).transformed[0] as ts.SourceFile;
+    }
+
+    const text = this.applyChanges(
+      this.printer.printFile(this.sourceFile),
+      changes
+    );
+    // clean source of new line placeholders
+    return text.replace(
+      new RegExp(
+        `(\r?\n)\\s*?${Util.escapeRegExp(NEW_LINE_PLACEHOLDER)}(\r?\n)`,
+        'g'
+      ),
+      `$1$2`
+    );
+  }
+
+  /**
+   * The format options to use when printing the source file.
+   */
+  public get formatOptions(): IFormatSettings {
+    return Object.assign(
+      {},
+      this._defaultFormatSettings,
+      this._formatSettingsFromConfig,
+      this.formatSettings
+    );
+  }
+
+  /**
+   * The language service host used to access the source file.
+   */
+  private _languageServiceHost: ts.LanguageServiceHost | undefined;
+  private get languageServiceHost(): ts.LanguageServiceHost {
+    if (!this._languageServiceHost) {
+      this._languageServiceHost = this.createLanguageServiceHost();
+    }
+
+    return this._languageServiceHost;
+  }
+
+  /**
+   * The language service instance used to format the source file.
+   */
+  private get languageService(): ts.LanguageService {
+    return ts.createLanguageService(
+      this.languageServiceHost,
+      ts.createDocumentRegistry()
+    );
+  }
+
+  /**
+   * Create a language service host for the source file.
+   * The host is used by TS to access the FS and read the source file.
+   * In this case we are operating on a single source file so we only need to provide its name and contents.
+   */
+  private createLanguageServiceHost(): ts.LanguageServiceHost {
+    const servicesHost: ts.LanguageServiceHost = {
+      getCompilationSettings: () => ({}),
+      getScriptFileNames: () => [this.sourceFile.fileName],
+      getScriptVersion: (_fileName) => '0',
+      getScriptSnapshot: (_fileName) => {
+        return ts.ScriptSnapshot.fromString(
+          this.printer.printFile(this.sourceFile)
+        );
+      },
+      getCurrentDirectory: () => process.cwd(),
+      getDefaultLibFileName: (options) => ts.getDefaultLibFilePath(options),
+      readDirectory: ts.sys.readDirectory,
+      readFile: ts.sys.readFile,
+      fileExists: ts.sys.fileExists,
+    };
+    return servicesHost;
+  }
+
+  /**
+   * Transform string literals to use single quotes.
+   */
+  private convertQuotesTransformer =
+    <T extends ts.Node>(context: ts.TransformationContext) =>
+    (rootNode: T): ts.Node => {
+      const visit = (node: ts.Node): ts.Node => {
+        if (ts.isStringLiteral(node)) {
+          const text = node.text;
+          // the ts.StringLiteral node has a `singleQuote` property that's not part of the public APi for some reason
+          // to make our lives easier we can modify it though
+          const singleQuote = (node as any).singleQuote;
+          const newNode = context.factory.createStringLiteral(text);
+          (newNode as any).singleQuote =
+            singleQuote || this.formatOptions.singleQuotes;
```

Comment:
Nope. The fact this is not exposed should say something.

Also, notice how the old logic only formats `createdStringLiterals`, i.e. the literals _we_ added to the source, the rest is _not_ the responsibility of the transform.
Also also, funnily enough, that was likely before `createStringLiteral` had a second param to specify that? Because we do have it used in places and it does work:
![image](https://github.com/IgniteUI/igniteui-cli/assets/3198469/82ad793f-2cc3-4a1d-95b3-55a05a480645)
---
So, how about we create the literals right the first time and skip the entire formatting quotes bit. Should also allow you to drop the transform step and the printer

Reply:
This is a good idea, but it would require that the transformer knows about the formatting options and has access to the file system (to get `.editorconfig/eslint` options) which is precisely what I want to avoid as an AST transformer should be concerned with the AST only and not the file system. I could do what we used to be doing - keep track of all the literals that we create and then replace them in the formatter based on the format options or I could try to figure out a different way to implement the transformer.